### PR TITLE
Improve dialyzer

### PR DESCRIPTION
**Pull request by michalwski**
This PR fixes remaining dialyzer warnings.
Also the job fails if dialyzer exits with status different then 0.

### CODE REVIEW

File: src/escalus_session.erl [hunk 1]
```diff
@@ -347,7 +347,7 @@ get_server_caps(Features) ->
     end.
 
 
--spec stream_start_to_element(exml_stream:start() | exml:element()) -> exml:element().
+-spec stream_start_to_element(exml_stream:element()) -> exml:element().
 stream_start_to_element(#xmlel{name = <<"open">>} = Open) -> Open;
 stream_start_to_element(#xmlstreamstart{name = Name, attrs = Attrs}) ->
```
Comment (gadgetci) [inline src/escalus_session.erl]: According to **Dialyzer**:
> Matching of pattern {'xmlstreamstart', Name, Attrs} tagged with a record name violates the declared type of #xmlel{name::<<_:32>>,attrs::[{_,_}],children::[{_,_} | {_,_,_,_}]}

File: src/escalus_history_h.erl [hunk 2]
```diff
@@ -10,11 +10,8 @@
...
 
--type dict_t() :: dict:dict().
-
 -record(state, {
-        events :: list(),
-        counters :: dict_t()
+        events :: list()
 }).
 
 get_history(Mgr) ->
```
Comment (gadgetci) [inline src/escalus_history_h.erl]: According to **Compiler**:
> Warning: missing specification for function get_history/1

**Comment by michalwski** [comment]
This was tested with MongooseIM here: michalwski/MongooseIM#64. All jobs are green :)

**Comment by michalwski** [comment]
Please ignore the failing `gadget/dialyzer` check I tried it but at this point it's not usable for us.

File: src/escalus.erl [hunk 3]
```diff
@@ -86,33 +90,44 @@ end_per_testcase(_CaseName, Config) ->
...
     escalus_users:delete_users(Config).
 
+-spec delete_users(config(), escalus_users:user_spec()) -> config().
 delete_users(Config, Users) ->
     escalus_users:delete_users(Config, Users).
 
+-spec get_users(Spec) -> Result when
+      Spec :: all | [escalus_users:user_name()]
+              | {by_name, [escalus_users:user_name()]},
+      Result :: [escalus_users:named_user()].
```
Comment (chrzaszcz) [inline src/escalus.erl]: minor: argument names should be the same in the spec and the function itself

Comment (michalwski) [inline src/escalus.erl]: Fixed in 3012667

**Pull request merged**